File: plugins/pandi-launchpad/src/pandi_launchpad/hooks.py
from __future__ import annotations

_EVENTS: dict[str, tuple[str, str]] = {
    "done": ("green", "static"),
    "attention": ("red", "pulse"),
}


def resolve_event(kind: str) -> tuple[str, str]:
    """Map a hook event kind to a (colour, mode) pair for lighting the whole grid."""
    return _EVENTS.get(kind, ("white", "flash"))

File: plugins/pandi-launchpad/src/hooks.ts
import type { Mode } from "./protocol.ts";

const EVENTS: Record<string, { color: string; mode: Mode }> = {
  done: { color: "green", mode: "static" },
  attention: { color: "red", mode: "pulse" },
};

export function resolveEvent(kind: string): { color: string; mode: Mode } {
  return EVENTS[kind] ?? { color: "white", mode: "flash" };
}
